Create a keystore needs additional information

### Issue

There are multiple versions of `keytool.exe` installed with Visual Studio 2022 (17.5.3, clean install, no other Android-related tools installed except those installed by Visual Studio Installer) but when a user opens the “Terminal” (View > Terminal), the default path relies upon the `keytool.exe` in the `Microsoft` folder – see below:

```
File Name	Path									File Size	Date Created	Date Accessed	Date Modified
keytool.exe	C:\Program Files\Android\jdk\jdk-8.0.302.8-hotspot\jdk8u302-b08\bin	19,664		3/16/2023 12:29	3/30/2023 12:44	3/16/2023 12:29
keytool.exe	C:\Program Files\Android\jdk\jdk-8.0.302.8-hotspot\jdk8u302-b08\jre\bin	19,664		3/16/2023 12:29	3/30/2023 12:44	3/16/2023 12:29
keytool.exe	C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\bin		19,664		7/20/2021 21:58	3/30/2023 12:40	7/20/2021 21:58
keytool.exe	C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\jre\bin	19,664		7/20/2021 22:01	3/30/2023 12:44	7/20/2021 22:01
keytool.exe	C:\Program Files\Microsoft\jdk-11.0.16.101-hotspot\bin			22,952		8/20/2022 21:42	3/30/2023 12:44	8/20/2022 21:42
```

```
PS E:\Source\GitHub\NationalParks> (Get-Command keytool.exe). Path
C:\Program Files\Microsoft\jdk-11.0.16.101-hotspot\bin\keytool.exe
```
But, that version of `keytool.exe` in the `Microsoft` folder does **NOT** prompt for a `key password` at the end – only for a `keystore password` at the beginning. All the other versions **DO** prompt for both `keystore password` and the beginning and `key password` at the end.

This documentation correctly instructs the user to include this in the .csproj file:

```
<PropertyGroup Condition="$(TargetFramework.Contains('-android')) and '$(Configuration)' == 'Release'">
    <AndroidKeyStore>True</AndroidKeyStore>
    <AndroidSigningKeyStore>myapp.keystore</AndroidSigningKeyStore>
    <AndroidSigningKeyAlias>key</AndroidSigningKeyAlias>
    <AndroidSigningKeyPass></AndroidSigningKeyPass>
    <AndroidSigningStorePass></AndroidSigningStorePass>
</PropertyGroup>
```
And to publish using this command:
`dotnet publish -f:net7.0-android -c:Release /p:AndroidSigningKeyPass=mypassword /p:AndroidSigningStorePass=mypassword`

But, if the 'keytool.exe' in the 'Microsoft' folder is used, the user will have no value to provide for the `key password` -- the first parameter in the command.  Without providing that value, the publish command fails with this error:

> C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.26\tools\Xamarin.Android.Common.targets(2323,2): error MSB4044: The “AndroidSignPackage” task was not given a value for the required parameter “KeyPass”
> 

### What needs done?

The documentation for "Create a keystore" needs to inform the user to **NOT** use the `keytool.exe` from the `Microsoft` folder and to explicitly invoke one of the others instead; e.g., those in one of the `Eclipse` or `Android` folders.


---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 6b6c3b8a-6109-5ad5-ca05-fdaae8a9d66e
* Version Independent ID: 6b6c3b8a-6109-5ad5-ca05-fdaae8a9d66e
* Content: [Use the CLI to publish for Android - .NET MAUI](https://learn.microsoft.com/en-us/dotnet/maui/android/deployment/publish-cli?view=net-maui-7.0)
* Content Source: [docs/android/deployment/publish-cli.md](https://github.com/dotnet/docs-maui/blob/main/docs/android/deployment/publish-cli.md)
* Product: **dotnet-mobile**
* Technology: **dotnet-maui**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**


---
[Associated WorkItem - 89017](https://dev.azure.com/msft-skilling/Content/_workitems/edit/89017)

--- a/docs/android/deployment/publish-cli.md
+++ b/docs/android/deployment/publish-cli.md
@@ -6,7 +6,7 @@ ms.date: 05/16/2023
 
 # Publish an Android app using the command line
 
-To distribute a .NET Multi-platform App UI (.NET MAUI) Android app, you'll need to sign it with a key from your keystore. A *keystore* is a database of security certificates that's created by using `keytool` from the Java SDK. A keystore is required when publishing a .NET MAUI Android app, as Android won't run apps that haven't been signed.
+To distribute a .NET Multi-platform App UI (.NET MAUI) Android app, you'll need to sign it with a key from your keystore. A *keystore* is a database of security certificates that's created by using `keytool` from the Java Development Kit (JDK). A keystore is required when publishing a .NET MAUI Android app, as Android won't run apps that haven't been signed.
 
 ## Create a keystore file
 
@@ -25,6 +25,9 @@ Perform the following steps to create a keystore file:
     keytool -genkeypair -v -keystore {filename}.keystore -alias {keyname} -keyalg RSA -keysize 2048 -validity 10000
     ```
 
+    > [!IMPORTANT]
+    > If you have multiple versions of the JDK installed on your computer, ensure that you run `keytool` from the latest version of the JDK.
+
     You'll be prompted to provide and confirm a password, followed by your full name, organization unit, organization, city or locality, state or province, and country code. This information is not displayed in your app, but is included in your certificate.
 
     For example, to generate a *myapp.keystore* file in the same folder as your project, with an alias of `myapp`, use the following command:
@@ -33,8 +36,8 @@ Perform the following steps to create a keystore file:
     keytool -genkeypair -v -keystore myapp.keystore -alias myapp -keyalg RSA -keysize 2048 -validity 10000
     ```
 
-    > [!IMPORTANT]
-    > Ensure you backup your keystore and password. If you lose it you'll be unable to sign your app with the same signing identity.  
+    > [!TIP!]
+    > Backup your keystore and password. If you lose it you'll be unable to sign your app with the same signing identity.  
 
 ## Find your keystore's signature
 

--- a/docs/android/deployment/publish-cli.md
+++ b/docs/android/deployment/publish-cli.md
@@ -36,7 +36,7 @@ Perform the following steps to create a keystore file:
     keytool -genkeypair -v -keystore myapp.keystore -alias myapp -keyalg RSA -keysize 2048 -validity 10000
     ```
 
-    > [!TIP!]
+    > [!TIP]
     > Backup your keystore and password. If you lose it you'll be unable to sign your app with the same signing identity.  
 
 ## Find your keystore's signature

